### PR TITLE
Add support for ceiling lights command

### PR DESCRIPTION
**Pull request by jklausa**
Hey there! :)

First, I wanted to thank you for working on the SwitchBot Cloud Home Assistant integration — it made my life much easier :)

However, I have some devices I'm using that aren't supported by it (namely the Ceiling Light Pro), so I took a stab at implementing those integrations myself — and this is the API part of it.

I also added a `DeviceOffline`, to be able to handle situation where you try to toggle a switch that's fallen off the network a bit more cleanly from HASS side — currently it just throws a "unknown error occured" in the UI.

I'm still working on and need to clean up the HASS part itself, but if you're curious, you can take a peek here: https://github.com/home-assistant/core/compare/dev...jklausa:core:switchbot-cloud-ceiling-light

If you have any feedback/suggestions, I'm more than open to it — I haven't written Python in like... 10 years, so my instincts are more than a little bit rusty :)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "switchbot_api"
-version = "1.3.0"
+version = "1.4.0"
```
Comment (SeraphicRav) [inline pyproject.toml]:
```suggestion
version = "2.0.0"
```
I think it a version as the error handling need to be updated

File: switchbot_api/__init__.py [hunk 2]
```diff
@@ -189,12 +202,21 @@ async def _request(self, path: str = "", callback: str = "get", json=None):
...
-                    raise CannotConnect()
-                return body.get("body")
 
+                if response.status >= 400:
+                    raise CannotConnect()
+                    
+                match body.get("statusCode"):
+                    case 100:
+                        return body.get("body")
+                    case 171:
```
Comment (SeraphicRav) [inline switchbot_api/__init__.py]: I think for the device, it should be 161, 171 is for an offline hub.

**Review by SeraphicRav — CHANGES_REQUESTED** [review]
@jklausa Thank so much for your PR ! I am really happy to see that this module and useful and that you want to contribute to it.

I am not considering myself a Python expert but I will do my best to review your code. The good looks clean to me.

I did a quick check on the Home Assistant Core side, I think you will need to do 3 PR. One to update this module version, another one for the plug sensors and another for the ceiling light.

Thanks again!

File: pyproject.toml (hunk 1 above)
Comment (jklausa) [reply pyproject.toml]: Good callout!

File: switchbot_api/__init__.py (hunk 2 above)
Comment (jklausa) [reply switchbot_api/__init__.py]: Hmm, in my testing my Plugs definitely throw 171 when they're disconnected from power. 

I'll try with my Ceiling Light Pro to see what that throws up...

Comment (jklausa) [reply switchbot_api/__init__.py]: Both the Ceiling Light, and the virtual remote devices also return 171 when offline. I've added 161 to match on both and a comment explaining it :)

Comment (SeraphicRav) [reply switchbot_api/__init__.py]: Oh I see, thanks for handling both cases and adding a comment explaining the situation :)

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "switchbot_api"
-version = "1.3.0"
+version = "2.0"
```
Comment (SeraphicRav) [inline pyproject.toml]:
```suggestion
version = "2.0.0"
```

**Review by SeraphicRav — DISMISSED** [review]
It looks good to me but the linter does not seem happy.

**Review by SeraphicRav — DISMISSED** [review]

**Comment by jklausa** [comment]
Sorry for that, should be good to go now!

**Review by SeraphicRav — APPROVED** [review]
OK, all green ! Thanks for you contribution.

**Comment by jklausa** [comment]
Fantastic :)! 

@SeraphicRav mind also bumping the release on PyPI?

**Comment by SeraphicRav** [comment]
@jklausa Oh thanks for the reminder, sorry about that. It is now published.

**Pull request merged**